core/TypeErrorDiagnostics.cc

```diff
@@ -66,21 +67,22 @@ void TypeErrorDiagnostics::maybeAutocorrect(const GlobalState &gs, ErrorBuilder
         e.replaceWith(fmt::format("Wrap in `{}`", *gs.suggestUnsafe), loc, "{}({})", *gs.suggestUnsafe,
                       loc.source(gs).value());
     } else {
+        // Duplicate the current constraint, so that we don't record any additional constraints
+        // simply from trying to generate autocorrects, and then solve it so that each additional
+        // heuristic here doesn't interfere with later heuristics.
+        auto constr = constrOrig.deepCopy();
```

Comment:
Can we deref `constrOrig.deepCopy()` here so we don't need to deref `constr` everywhere it's used?

Reply:
We can't do it here, `constr` is a `unique_ptr`. If we deref'd it right here, it would be dropped after the line, and the memory pointed to could be overwritten with garbage.

We could do it on another line in a new variable:

```cpp
auto constrPtr = constrOrig.deepCopy();
auto constr = *constrPtr;
```

But that's more characters than typing than `*constr` everywhere (until there's another 20 `*constr` in this method), and even then the original variable would still be in scope, so it would be possible to accidentally still use `*constrPtr` instead of `constr`.

So IMO it's overkill, it's going to be in the CPU cache in all likelihood (and again, this method is not performance sensitive).